compaction/compaction.cc

```diff
@@ -625,11 +625,6 @@ class compaction {
 
     flat_mutation_reader_v2::filter make_partition_filter() const {
         return [this] (const dht::decorated_key& dk) {
-#ifdef SEASTAR_DEBUG
-            // sstables should never be shared with other shards at this point.
-            assert(dht::shard_of(*_schema, dk.token()) == this_shard_id());
-#endif
-
```

Comment:
How did you spot it? Should have been caught in some of the tests running a debug-mode binary. But I don't think longevity tests do it. Navigating through test dir, I see we miss a table_test, perhaps we can start one with a cleanup test.

Reply:
I ran a bunch of relevant dtests in debug mode.